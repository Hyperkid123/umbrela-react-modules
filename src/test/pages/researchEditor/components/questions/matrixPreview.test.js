import React from 'react'
import MatrixPreview from '../../../../../pages/researchEditorModule/components/questions/matrixPreview';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Matrix Preview component', () => {
  it('should render correctly empty', () => {
    const tree = shallow(
      <MatrixPreview
        options={[]}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render correctly with options', () => {
    const tree = shallow(
      <MatrixPreview
        questionType='MatrixSingleImageQuestion'
        options={[{
          optionType: 'ColumnOption',
          optionId: 0,
        }, {
          optionType: 'RowOption',
          optionId: 1,
        }]}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render checkboxes', () => {
    const tree = shallow(
      <MatrixPreview
        questionType='MatrixMulti'
        options={[{
          optionType: 'ColumnOption',
          optionId: 0,
        }, {
          optionType: 'RowOption',
          optionId: 1,
        }]}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should close image preview on new props', () => {
    const wrapper = shallow(
      <MatrixPreview
        options={[]}
      />
    );
    wrapper.setProps({});
    expect(wrapper.state().modalOpen).toBeFalsy();
  });

  it('should open modal with image preview', () => {
    const wrapper = shallow(
      <MatrixPreview
        options={[]}
      />
    );
    wrapper.instance().handleOpen('foo.png');
    expect(wrapper.state().modalOpen).toBeTruthy();
    expect(wrapper.state().imgSource).toEqual('foo.png');
  })

  it('should close modal with image preview', () => {
    const wrapper = shallow(
      <MatrixPreview
        options={[]}
      />
    );
    wrapper.instance().handleClose();
    expect(wrapper.state().modalOpen).toBeFalsy();
  })
});
