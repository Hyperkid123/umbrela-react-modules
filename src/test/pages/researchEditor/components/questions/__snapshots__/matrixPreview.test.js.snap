// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matrix Preview component should render checkboxes 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.table>
    <styled.thead>
      <tr>
        <styled.td
          key="empty"
        />
        <styled.td
          filler={true}
          key="0"
        >
          <span />
        </styled.td>
      </tr>
    </styled.thead>
    <styled.tbody>
      <styled.tr
        filler={true}
        key="1"
      >
        <styled.td>
          <span />
        </styled.td>
        <styled.td
          key="0"
        >
          <WithStyles
            checkedIcon={<ShouldUpdate />}
            icon={<ShouldUpdate />}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
      </styled.tr>
    </styled.tbody>
  </styled.table>
  <WithStyles
    actions={
      Array [
        <WithStyles
          label="Zavřít"
          onClick={[Function]}
          primary={true}
      />,
      ]
    }
    autoScrollBodyContent={true}
    contentStyle={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
        "maxWidth": "none",
        "width": "90%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={false}
    title="Náhled"
  >
    <LazyLoad
      offset={0}
      once={false}
      overflow={false}
      resize={false}
      scroll={true}
      unmountIfInvisible={false}
    >
      <styled.img
        alt={null}
        src={null}
      />
    </LazyLoad>
  </WithStyles>
</div>
`;

exports[`Matrix Preview component should render correctly empty 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.table>
    <styled.thead>
      <tr>
        <styled.td
          key="empty"
        />
      </tr>
    </styled.thead>
    <styled.tbody />
  </styled.table>
  <WithStyles
    actions={
      Array [
        <WithStyles
          label="Zavřít"
          onClick={[Function]}
          primary={true}
      />,
      ]
    }
    autoScrollBodyContent={true}
    contentStyle={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
        "maxWidth": "none",
        "width": "90%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={false}
    title="Náhled"
  >
    <LazyLoad
      offset={0}
      once={false}
      overflow={false}
      resize={false}
      scroll={true}
      unmountIfInvisible={false}
    >
      <styled.img
        alt={null}
        src={null}
      />
    </LazyLoad>
  </WithStyles>
</div>
`;

exports[`Matrix Preview component should render correctly with options 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.table>
    <styled.thead>
      <tr>
        <styled.td
          key="empty"
        />
        <styled.td
          filler={true}
          key="0"
        >
          <span />
        </styled.td>
      </tr>
    </styled.thead>
    <styled.tbody>
      <styled.tr
        filler={true}
        key="1"
      >
        <styled.td>
          <styled.img
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          />
        </styled.td>
        <styled.td
          key="0"
        >
          <WithStyles
            checkedIcon={<ShouldUpdate />}
            icon={<ShouldUpdate />}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
      </styled.tr>
    </styled.tbody>
  </styled.table>
  <WithStyles
    actions={
      Array [
        <WithStyles
          label="Zavřít"
          onClick={[Function]}
          primary={true}
      />,
      ]
    }
    autoScrollBodyContent={true}
    contentStyle={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
        "maxWidth": "none",
        "width": "90%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={false}
    title="Náhled"
  >
    <LazyLoad
      offset={0}
      once={false}
      overflow={false}
      resize={false}
      scroll={true}
      unmountIfInvisible={false}
    >
      <styled.img
        alt={null}
        src={null}
      />
    </LazyLoad>
  </WithStyles>
</div>
`;
